src/fenced/fenced/disks.py

```diff
@@ -97,7 +98,7 @@ def callback(i, fs, failed):
             except Exception:
                 failed.add(fs[i])
 
-        failed = self._run_batch('get_keys', done_callback=callback)
+        remote_keys = self._run_batch('get_keys', done_callback=callback)
```

Comment:
That doesnt seem right, _run_batch is supposed to return failures, not get_keys values

Reply:
The reason why is because the `callback` is making a check to see if the `host_key` is None and if it is it will add the disk to the `failed` set and then return.

In this scenario, host_key will always be `None` because the other controller has the reservation keys set.

Reply:
and why host_key is None? I dont see how/why it would be.

Even if it can be this does not seem like the right way to handle this as its not a failure and we will be adding it there.
It will mix actual failures with remote_keys results.

Reply:
Spotted the actual problem. 👍 PR updated.